utils.lua

```diff
@@ -150,4 +150,16 @@ function nn.utils.addSingletonDimension(t, dim)
   return view
 end
 
+function nn.utils.contiguousView(output, input, ...)
+  output = output or input.new()
+  if input:isContiguous() then
+    output:view(input, ...)
+  else
+    output:resizeAs(input)
+    output:copy(input)
+    output:view(...)
```

Comment:
Sorry, just noticed that you want to put output as src in your second suggestion. I think this output:view(output,...) 
should not be necessary and is a tiny bit less readable.